### ISSUE TITLE
[Bug]: Tooltip/Icon only button on an sortable DataTable header 

### ISSUE THREAD

**Issue by laura-genari** [carbon-design-system/carbon#15403]
### Package

@carbon/react, @carbon/icons-react

### Browser

Chrome, Firefox

### Package version

1.45.0

### React version

17.0.1

### Description

HI team. So, I am having this warning on devtools `Warning: validateDOMNesting(...): <button> cannot appear as a descendant of <button>`. because I have a tooltip inside a sortable header on DataTable. The tooltip component is actually an icon only button, but I tried using Tooltip and DefinitionTooltip, but nothing works. 

### Reproduction/example

https://stackblitz.com/edit/github-bi6asm?file=src%2FApp.jsx

### Steps to reproduce

- Create a sortable DataTable
- Put a tooltip in a cell header

### Suggested Severity

Severity 3 = User can complete task, and/or has a workaround within the user experience of a given component.

### Application/PAL

zPricer/WLP

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by tw15egan** [comment]
I'm not sure this will be feasible a11y-wise; since the sortable data table is an interactive element, you are not supposed to nest another interactive element inside of it.

**Comment by laura-genari** [comment]
So, this tooltip is for explaining what an acronym is (which was a compliance requirement). What should I do in this case?

**Comment by tay1orjones** [comment]
Usually how we approach this is to render it adjacent/outside the button and then absolutely positioned it atop the other button (in this case the header).
